### PR TITLE
Add Nexus Lifecycle Support

### PR DESCRIPTION
**Pull request by DarthHater**
Welp, this here PR adds:

* `iq` command to `bach`
* a `CycloneDX` class that will create an sbom given a list of purls
* a `IQClient` class that handles talking to IQ Server
* tests for all public methods

Works locally testing against my local Nexus IQ Server

### CODE REVIEW

File: app/Commands/IQ.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+        $cyclonedx = new CycloneDX();
+
+        $sbom = $cyclonedx->create_and_return_sbom($coordinates);
+
+        $iq_client = new IQClient(null, $this->option('host'), $this->option('user'), $this->option('token'), $this->option('stage'));
+
+        $internal_id = $iq_client->get_internal_application_id($this->option('application'));
+
+        $status_url = $iq_client->submit_sbom($sbom, $internal_id, 'develop');
```
Comment (bhamail) [inline app/Commands/IQ.php:78]: So begins the start of my silly questions: Should the `develop` parameter be hardcoded here? Or should it be something like: `$this->option('develop')`

File: app/IQ/IQClient.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+     * The passed in value must be a Guzzle Client
+     *
+     * @param Client|null $client
+     */
+    public function __construct(
+        $client = null,
+        $host = 'http://localhost:8070/',
+        $username = 'admin',
+        $token = 'admin123',
+        $stage = 'develop'
```
Comment (bhamail) [inline app/IQ/IQClient.php:28]: To my clueless eyes, it appears this field is not used, but instead the `$iq_client->submit_sbom(..., stage)` is used? Maybe one of these should be removed.

**Review by bhamail — COMMENTED** [review]
Still getting up to speed, but wanted to share some initial silly questions.

File: app/Commands/IQ.php (hunk 1 above)
Comment (DarthHater) [reply app/Commands/IQ.php:78]: No, you are correct!

File: app/IQ/IQClient.php (hunk 2 above)
Comment (DarthHater) [reply app/IQ/IQClient.php:30]: Addressed in: dd0b2b7

File: app/Commands/IQ.php (hunk 1 above)
Comment (DarthHater) [reply app/Commands/IQ.php]: Addressed in: dd0b2b7

**Pull request merged**